File: StudyFlowBackend/StudyFlowBackend/Models/Entities/UserNotification.cs
using System;

namespace StudyFlowBackend.Models
{
    public class UserNotification
    {
        public Guid Id { get; set; }
        
        public string UserId { get; set; } = string.Empty;
        public User? User { get; set; }
        
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public bool IsRead { get; set; } = false;
        
        /// <summary>
        /// Loại thông báo: "Daily" (nhắc nhở ngày) hoặc "CustomTask" (nhắc nhở cụ thể công việc)
        /// </summary>
        public string Type { get; set; } = "General";
    }
}

File: lib/features/notification/domain/entities/user_notification.dart
/// Entity representing a notification for the user.
class UserNotification {
  final String id;
  final String userId;
  final String title;
  final String message;
  final DateTime createdAt;
  final bool isRead;
  final String type;

  UserNotification({
    required this.id,
    required this.userId,
    required this.title,
    required this.message,
    required this.createdAt,
    required this.isRead,
    required this.type,
  });
}
